src/gc.c

```diff
@@ -2461,6 +2458,10 @@ void jl_gc_init(void)
     last_long_collect_interval = default_collect_interval;
     allocd_bytes = -default_collect_interval;
 
+    char *max_pages_str = getenv("GC_MAX_PAGES");
```

Comment:
Make this `JULIA_GC_MAX_PAGES`. While at it, should probably change `JL_GC_NO_GENERATIONAL` to `JULIA_GC_NO_GENERATIONAL`